src/core/core.js

```diff
@@ -233,6 +233,7 @@ export default class Core {
         if (typeof analyticsEvent === 'object') {
           this._analyticsReporter.report(AnalyticsEvent.fromData(analyticsEvent));
         }
+        this.globalStorage.set(StorageKeys.VERTICAL_SEARCH_LOADED, true);
```

Comment:
I might be off-base here, but this seems a bit roundabout to me. We need the `AppliedFiltersComponent` to `setState` only after the `FilterOptionsComponent` has set the new Facets in the registry. Should we set a key in `globalStorage` in that component after it makes the call? It feels like we're assuming that `FilterOptions` will make the update by this point in the code. 

I'm a bit wary of the key name too. It sounds similar to something we have already. `VerticalResults` has a `searchState` that can be `SEARCH_LOADING`. I wouldn't want people to get confused about two similarly named things.

Lastly, do the `updateListeners` get invoked whenever a `globalStorage.set` is called on their `StorageKey`? I'm assuming we don't do a comparison first to see if the value's changed before invoking the listeners. If we do, I guess we'd need to reset this value.

Reply:
We do a shallow equality comparison, so we don't need to reset this value. I think a new key in the component makes more sense for the specific bug we're trying to fix!

Reply:
I updated things so that Facets component sets FACETS_LOADED at the end of it's onMount. The problem with this is that the AppliedFilters component would then have to listen to both FACETS_LOADED and VERTICAL_RESULTS, otherwise it wouldn't work properly without a Facets component on the page. This results in 2 setStates per query if you're using Facets (verified with console logs). Visually I couldn't see any flickering behavior at all, it feels a little silly, but I guess it's not the worst. 

Another alternative would be flipping the order VERTICAL_RESULTS and DYNAMIC_FILTERS are updated. I'm pretty sure this would not have any side effects (did some quick testing and looks like it doesn't), but it's a little scary.

Anyways, I feel like either we would want to find a better name for another StorageKey is updated after DYNAMIC_FILTERS during a VerticalSearch, or live with the double setState

Reply:
Could you explain a bit more how `AppliedFilters` will work if the `FacetsComponent` is not present? I got the impression that the `FacetsComponent` was needed for `AppliedFilters` to work, since it added facets to the registry. If the `AppliedFilters` component works fine on it's own (when `FacetsComponent` is not around), I don't totally follow why having them together is an issue.

Reply:
`AppliedFilters` just pulls from all the filters in FilterRegistry, it doesn't care about the difference between facet filters vs static filters. The issue is, currently the SDK relies on `Facets` to register a new search's facet filters into FilterRegistry. If the `AppliedFilters` setState happens before `Facets` setState, `AppliedFilters` will pull all FilterNodes from `FilterRegistry` before `Facets` has a chance to register them, which ends up pulling all the facet FilterNodes sent in the request, not the ones parsed from the response. This leads to issues when you run a search like "consulting", which does not send any facets in the request, but will return a response with the "consulting" facet selected.

Reply:
If `Facets` are not present, `AppliedFilters` will still pull all filter nodes from `FilterRegistry` as normal, which handles static filters properly since static filters don't change when they come back.

Reply:
Because we rely on the `Facets` component to register the new dynamic filters, can `AppliedFilters` show these filters if `Facets` has not been added to the page?

Reply:
But it can't show any of the dynamic filters that were applied without `Facets`? If so, that seems like an issue and I bet Product expects that to work.

Reply:
Nope, without a `Facets` component no Facets will be added to the FilterRegistry. If we move the creation of FilterNodes for facets outside of the `Facets` component they would be, though.

Reply:
We should tell Rose about this. She should hear it from us and not a TechOps.

Reply:
If we can quickly come up with a sane way to avoid the double `setState`, I say we should. If it seems like that will be difficult or time consuming, I'm okay leaving it for now. We already know that the SDK's filters are in terrible shape and are planning an imminent refactor. We don't want to hold up 1.6 over this.